misc/python/materialize/version_consistency/version_consistency_test.py

```diff
@@ -0,0 +1,141 @@
+    MultiVersionSqlExecutors,
+)
+from materialize.version_consistency.ignore_filter.version_consistency_ignore_filter import (
+    VersionConsistencyIgnoreFilter,
+)
+
+
+class VersionConsistencyTest(OutputConsistencyTest):
+    def __init__(self) -> None:
+        self.mz2_connection: Connection | None = None
+        self.evaluation_strategy_name: str | None = None
+
+    def get_scenario(self) -> EvaluationScenario:
+        return EvaluationScenario.VERSION_CONSISTENCY
+
+    def create_sql_executors(
+        self,
+        config: ConsistencyTestConfiguration,
+        connection: Connection,
+        output_printer: OutputPrinter,
+    ) -> SqlExecutors:
+        assert self.mz2_connection is not None, "Second connection is not initialized"
+
+        return MultiVersionSqlExecutors(
+            create_sql_executor(config, connection, output_printer, "mz1"),
+            create_sql_executor(config, self.mz2_connection, output_printer, "mz2"),
+        )
+
+    def create_result_comparator(
+        self, ignore_filter: InconsistencyIgnoreFilter
+    ) -> ResultComparator:
+        return ResultComparator(ignore_filter)
+
+    def create_inconsistency_ignore_filter(
+        self, sql_executors: SqlExecutors
+    ) -> InconsistencyIgnoreFilter:
+        assert isinstance(sql_executors, MultiVersionSqlExecutors)
+        return VersionConsistencyIgnoreFilter(
+            sql_executors.executor.query_version(),
+            sql_executors.executor2.query_version(),
+        )
+
+    def create_evaluation_strategies(self) -> list[EvaluationStrategy]:
+        assert (
+            self.evaluation_strategy_name is not None
+        ), "Evaluation strategy name is not initialized"
+
+        strategies: list[EvaluationStrategy]
+
+        if self.evaluation_strategy_name == "dataflow_rendering":
+            strategies = [DataFlowRenderingEvaluation(), DataFlowRenderingEvaluation()]
+            strategies[1].identifier = EvaluationStrategyKey.MZ_DATAFLOW_RENDERING_DB_2
+        elif self.evaluation_strategy_name == "constant_folding":
+            strategies = [ConstantFoldingEvaluation(), ConstantFoldingEvaluation()]
+            strategies[1].identifier = EvaluationStrategyKey.MZ_CONSTANT_FOLDING_DB_2
+        else:
+            raise RuntimeError(f"Unexpected name: {self.evaluation_strategy_name}")
+
+        for i, strategy in enumerate(strategies):
+            number = i + 1
+            strategy.name = f"{strategy.name} {number}"
+            strategy.object_name_base = f"{strategy.object_name_base}_{number}"
+            strategy.simple_db_object_name = (
+                f"{strategy.simple_db_object_name}_{number}"
+            )
+
+        return strategies
+
+
+def main() -> int:
+    test = VersionConsistencyTest()
+    parser = argparse.ArgumentParser(
+        prog="postgres-consistency-test",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="Test the consistency of Materialize and Postgres",
```

Comment:
This comment may no longer be appropriate if this script is comparing two Mz versions.

Reply:
Thanks, fixed.